fabric-item-api-v1/src/main/java/net/fabricmc/fabric/mixin/item/AbstractFurnaceBlockEntityMixin.java

```diff
@@ -0,0 +1,33 @@
+package net.fabricmc.fabric.mixin.item;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.block.entity.AbstractFurnaceBlockEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.collection.DefaultedList;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+@Mixin(AbstractFurnaceBlockEntity.class)
+public class AbstractFurnaceBlockEntityMixin {
+	private static DefaultedList<ItemStack> capturedInventory;
+
+	@Inject(method = "tick", at = @At("HEAD"))
+	private static void getStackCraftingRemainder(World world, BlockPos pos, BlockState state, AbstractFurnaceBlockEntity blockEntity, CallbackInfo ci) {
+		capturedInventory = ((AbstractFurnaceBlockEntityAccessor)blockEntity).getInventory();
```

Comment:
Can't you shadow since it's in the same class?

Reply:
No, its static and you can't capture locals on redirects/modify var

Reply:
I don't see how the annotation type matters? This should work afaik:
```java
@Shadow private Whatever inventory;

@Inject(...)
private ...() {
    capturedInventory = blockEntity.inventory;
}
```

Reply:
![image](https://user-images.githubusercontent.com/26313415/184510930-48433d00-e9fc-4826-96b4-61a8e87b886f.png)
Do you mean something like this?

Reply:
Yeah, I think that should work. It looks a bit cleaner than a separate accessor just for a mixin to the same class.